File: Insert_Intervals/insert_intervals.py
from typing import List


class Solution:
    def insert(
        self, intervals: List[List[int]], newInterval: List[int]
    ) -> List[List[int]]:
        merge_intervals: List[List[int]] = []

        index = 0

        while index < len(intervals) and intervals[index][1] < newInterval[0]:
            merge_intervals.append(intervals[index])
            index += 1

        while index < len(intervals) and intervals[index][0] <= newInterval[1]:
            newInterval[0] = min(newInterval[0], intervals[index][0])
            newInterval[1] = max(newInterval[1], intervals[index][1])
            index += 1

        merge_intervals.append(newInterval)

        while index < len(intervals):
            merge_intervals.append(intervals[index])
            index += 1

        return merge_intervals

File: Insert_Intervals/insert_intervals.js
/**
 * @param {number[][]} intervals - The array of intervals sort in asending order of first element
 * @param {number[]} newInterval - The new interval to insert
 * @returns {number[][]} returns the new intervals after insertion
 */
var insert = function (intervals, newInterval) {
	const result = [];
	let i = 0;

	while (i < intervals.length && intervals[i][1] < newInterval[0]) {
		result.push(intervals[i]);
		i++;
	}

	while (i < intervals.length && intervals[i][0] <= newInterval[1]) {
		newInterval[0] = Math.min(newInterval[0], intervals[i][0]);
		newInterval[1] = Math.max(newInterval[1], intervals[i][1]);
		i++;
	}

	result.push(newInterval);

	while (i < intervals.length) {
		result.push(intervals[i]);
		i++;
	}

	return result;
};
